src/serac/physics/state/state_manager.cpp

```diff
@@ -21,90 +21,99 @@ QuadratureData<void> dummy_qdata;
-        "parallel");
-  } else {
-    datacoll_->SetPrefixPath(output_directory);
-  }
+  datacoll.SetPrefixPath(output_dir_);
 
   if (cycle_to_load) {
-    is_restart_ = true;
     // NOTE: Load invalidates previous Sidre pointers
-    datacoll_->Load(*cycle_to_load);
-    datacoll_->SetGroupPointers(
-        ds.getRoot()->getGroup(collection_name_ + "_global/blueprint_index/" + collection_name_),
-        ds.getRoot()->getGroup(collection_name_));
-    SLIC_ERROR_ROOT_IF(datacoll_->GetBPGroup()->getNumGroups() == 0,
+    datacoll.Load(*cycle_to_load);
+    datacoll.SetGroupPointers(ds_->getRoot()->getGroup(coll_name + "_global/blueprint_index/" + coll_name),
+                              ds_->getRoot()->getGroup(coll_name));
+    SLIC_ERROR_ROOT_IF(datacoll.GetBPGroup()->getNumGroups() == 0,
                        "Loaded datastore is empty, was the datastore created on a "
                        "different number of nodes?");
 
-    datacoll_->UpdateStateFromDS();
-    datacoll_->UpdateMeshAndFieldsFromDS();
+    datacoll.UpdateStateFromDS();
+    datacoll.UpdateMeshAndFieldsFromDS();
 
     // Functional needs the nodal grid function and neighbor data in the mesh
     mesh().EnsureNodes();
     mesh().ExchangeFaceNbrData();
 
   } else {
-    datacoll_->SetCycle(0);   // Iteration counter
-    datacoll_->SetTime(0.0);  // Simulation time
+    datacoll.SetCycle(0);   // Iteration counter
+    datacoll.SetTime(0.0);  // Simulation time
+  }
+}
+
+void StateManager::initialize(axom::sidre::DataStore& ds, const std::string& output_directory)
+{
+  // If the global object has already been initialized, clear it out
+  if (ds_) {
+    reset();
+  }
+  ds_         = &ds;
+  output_dir_ = output_directory;
+  if (output_directory.empty()) {
+    SLIC_ERROR_ROOT(
+        "DataCollection output directory cannot be empty - this will result in problems if executables are run in "
+        "parallel");
   }
 }
 
-FiniteElementState StateManager::newState(FiniteElementVector::Options&& options)
+FiniteElementState StateManager::newState(FiniteElementVector::Options&& options, const std::string& tag)
 {
-  SLIC_ERROR_ROOT_IF(!datacoll_, "Serac's datacollection was not initialized - call StateManager::initialize first");
-  const std::string name = options.name;
+  SLIC_ERROR_ROOT_IF(!ds_, "Serac's datacollection was not initialized - call StateManager::initialize first");
+  auto&             datacoll = datacolls_.at(tag);
+  const std::string name     = options.name;
   if (is_restart_) {
-    auto field = datacoll_->GetParField(name);
+    auto field = datacoll.GetParField(name);
     return {mesh(), *field, name};
   } else {
-    SLIC_ERROR_ROOT_IF(datacoll_->HasField(name),
+    SLIC_ERROR_ROOT_IF(datacoll.HasField(name),
                        axom::fmt::format("Serac's datacollection was already given a field named '{0}'", name));
     options.managed_by_sidre = true;
     FiniteElementState state(mesh(), std::move(options));
```

Comment:
Should this `FiniteElementState` refer to the mesh associated with `tag`?

Reply:
Good catch, thanks!